### PR TITLE
Add better UI for different errors on note page

### PR DESCRIPTION
**Pull request by Eprince-hub**
Ensure that the UI that is being displayed to the users reflects the message that is coming back from the server.

When a user queries for a note that doesn't exist, we want to display a UI that shows this to the user instead of showing `Access Denied` in this case

### CODE REVIEW

File: app/notes/[noteId].tsx [hunk 1]
```diff
@@ -78,8 +77,8 @@ export default function Note() {
   return (
     <View style={styles.container}>
       <View style={styles.textContainer}>
-        <Text style={styles.title}>{note.title}</Text>
-        <Text style={styles.textContent}>{note.textContent}</Text>
+        <Text style={styles.title}>{note?.title}</Text>
+        <Text style={styles.textContent}>{note?.textContent}</Text>
```
Comment (karlhorky) [inline app/notes/[noteId].tsx:81]: this change doesn't look right - we shouldn't introduce optional chaining here

File: app/api/notes/[noteId]+api.ts [hunk 2]
```diff
@@ -30,9 +30,10 @@ export async function GET(
       },
     );
   }
-  const note = await getNote(token, Number(noteId));
 
-  if (!note) {
+  const noteExists = await selectNoteExists(Number(noteId));
+
+  if (!noteExists) {
```
Comment (karlhorky) [inline app/api/notes/[noteId]+api.ts:36]: since this is a 1-time usage variable, we can inline it in the `if` condition

Comment (Eprince-hub) [reply app/api/notes/[noteId]+api.ts]: 083f7f9c8d47c58fc2a88c08b78f66d7b041f7cb

File: app/notes/[noteId].tsx (hunk 1 above)
Comment (Eprince-hub) [reply app/notes/[noteId].tsx]: 36dde012dafc577628274c2899510c90f09456a7

**Review by karlhorky — APPROVED** [review]
LGTM, thanks!

**Pull request merged**